File: ExampleCompose/app/src/main/java/com/example/examplecompose/data/bus/BusModule.kt
package com.example.examplecompose.data.bus

import com.example.examplecompose.data.bus.remote.api.GetBusStopService
import com.example.examplecompose.data.bus.repository.GetBusStopRepositoryImpl
import com.example.examplecompose.data.common.module.RetrofitModule
import com.example.examplecompose.domain.bus.GetBusStopRepository
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import retrofit2.Retrofit
import javax.inject.Singleton

@Module(includes = [RetrofitModule::class])
@InstallIn(SingletonComponent::class)
class BusModule {
    @Singleton
    @Provides
    fun provideBusService(retrofit: Retrofit) : GetBusStopService {
        return retrofit.create(GetBusStopService::class.java)
    }

    @Singleton
    @Provides
    fun provideLoginRepository(busStopService: GetBusStopService) : GetBusStopRepository {
        return GetBusStopRepositoryImpl(busStopService)
    }
}

File: ExampleFlutter/example_flutter/lib/data/bus/bus_module.dart
import 'package:dio/dio.dart';
import 'package:example_flutter/data/bus/remote/api/bus_remote.dart';

BusStopRemote? busStopRemote;

BusStopRemote singletonBusStopClient(Dio dio) {
  busStopRemote ??= BusStopRemote(dio);

  return busStopRemote!;
}